dspace-api/src/main/java/org/dspace/versioning/VersioningConsumer.java

```diff
@@ -49,37 +73,392 @@ public void finish(Context ctx) throws Exception {
+                    // Set the new version of the item to latest if the relevant relationship exists (match found).
+                    // This implies that the new version of the item will appear on the page of the third-party item.
+                    // The old version of the item will not appear anymore on the page of the third-party item,
+                    // see previous step.
+                    relationshipVersioningUtils.updateLatestVersionStatus(latestItemRelationship, isLeft, true);
                 }
             }
         }
     }
 
+    /**
+     * Given two items, check if their entity types match.
+     * If one or both items don't have an entity type, comparing is pointless and this method will return false.
+     * @param latestItem the item that represents the most recent version.
+     * @param previousItem the item that represents the second-most recent version.
+     * @return true if the entity types of both items are non-null and equal, false otherwise.
+     */
+    protected boolean doEntityTypesMatch(Item latestItem, Item previousItem) {
+        String latestItemEntityType = getEntityType(latestItem);
+        String previousItemEntityType = getEntityType(previousItem);
+
+        // check if both items have an entity type
+        if (latestItemEntityType == null || previousItemEntityType == null) {
+            if (previousItemEntityType != null) {
+                log.warn(String.format(
+                    "Inconsistency: Item with uuid %s, handle %s has NO entity type, " +
+                    "but the previous version of that item with uuid %s, handle %s has entity type %s",
+                    latestItem.getID(), latestItem.getHandle(),
+                    previousItem.getID(), previousItem.getHandle(), previousItemEntityType
+                ));
+            }
+
+            // one or both items do not have an entity type, so comparing is pointless
+            return false;
+        }
+
+        // check if the entity types are equal
+        if (!StringUtils.equals(latestItemEntityType, previousItemEntityType)) {
+            log.warn(String.format(
+                "Inconsistency: Item with uuid %s, handle %s has entity type %s, " +
+                "but the previous version of that item with uuid %s, handle %s has entity type %s",
+                latestItem.getID(), latestItem.getHandle(), latestItemEntityType,
+                previousItem.getID(), previousItem.getHandle(), previousItemEntityType
+            ));
+            return false;
+        }
+
+        // success - the entity types of both items are non-null and equal
+        log.info(String.format(
+            "Item with uuid %s, handle %s and the previous version of that item with uuid %s, handle %s " +
+            "have the same entity type: %s",
+            latestItem.getID(), latestItem.getHandle(), previousItem.getID(), previousItem.getHandle(),
+            latestItemEntityType
+        ));
```

Comment:
Small note.. While this isn't "wrong", there's an easier way. I think you can just drop the `String.format` and turn this into:
`log.info("Item with uuid {}, handle {} ... ", latestItem.getId(), latestItem.getHandle(), ...);`

But, like I said, this code isn't wrong... it's just slightly more complex than it needs to be.  So, you can either correct these or leave them as-is.  This same is true for all your other log4j code that calls `String.format()`.... this isn't needed anymore for log4j v2.

Reply:
This has been updated where it was not causing complexities